### ISSUE TITLE
Training Network Heads and All Layer Training

### ISSUE THREAD

**Issue by khanfarhan10** [ayoolaolafenwa/PixelLib#57]
From the official [Mask-RCNN example](https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/samples/nucleus/nucleus.py#L279) it is apparent that two approaches are used :

- Train network heads
- Train all layers

Hence the question arises why is this needed?  

Are there specific use cases when either of them is used??

I understand that the [equivalent implementation in PixelLib](https://github.com/ayoolaolafenwa/PixelLib/blob/e6af2be69d12d06e307670ba55eda309ca463a8c/pixellib/custom_train.py#L103) is as follows :

```
train_model(num_epochs, path_trained_models,  layers = "all" or "head", augmentation)
```

**Comment by ayoolaolafenwa** [comment]
@khanfarhan10 The major reason is that training all layers learn more abstract features about the data compared to training just the network heads.  When the data is large, training it with all layers is best, while training network heads works best for a small dataset.

**Comment by khanfarhan10** [comment]
Ah great, it is definitely more clear now thanks.

But does it make sense to do both for a single dataset?

Like train `all` for `epochs=100` and then train `heads` for `epochs=200`?

**Comment by ayoolaolafenwa** [comment]
> Ah great, it is definitely more clear now thanks.
> 
> But does it make sense to do both for a single dataset?
> 
> Like train `all` for `epochs=100` and then train `heads` for `epochs=200`?

I have not tried it, but I will advise you to stick to training only all layers for best performance.
